Groups in worker

vegarringdal opened the pull request:
This should help this issue:
https://github.com/gkjohnson/three-mesh-bvh/issues/392#issuecomment-1072703011

gkjohnson reviewed (commented):
Thanks! Just added a comment -- I think we can do this automatically so it "just works". Appreciate the contribution!

vegarringdal commented on src/workers/generateAsync.worker.js:
```diff
@@ -43,6 +43,12 @@ global.onmessage = function ( { data } ) {
 
 		}
 
+		if ( options.groups ) {
+
+			geometry.groups = options.groups;
+
+		}
```
yes, more automatic might be nice.
didt really think about it since I use premerged meshed and have group info somewhere else. Only reason why I add it is to give it to mesh-bvh so I can later generate a "selection" mesh 😁

gkjohnson replied:
Great thanks for the change :)

vegarringdal commented:
@gkjohnson 
Did a update to check if geometry have groups, if it does I just send it over with options..
Also removed the comments I added to readme.

At least now its a little more automatic.

gkjohnson commented on src/workers/generateAsync.worker.js:
```diff
@@ -43,6 +43,12 @@ global.onmessage = function ( { data } ) {
 
 		}
 
+		if ( options.groups ) {
+
+			geometry.groups = options.groups;
+
+		}
```
Can we iterate over the geometry groups here and use the [geometry.addGroup](https://threejs.org/docs/?q=buffergeometry#api/en/core/BufferGeometry.addGroup) function here, instead? If the function changes, is deprecated, or has side effects added later on it would be best to go through the correct method.

vegarringdal replied:
ok, Ill try and have a look later when Im home.
Didnt know about the function 😂
Im lazy in my experiment and just added what I needed.

Noticed the material index, can we actually have different materials on merged meshes by using the groups ? Do you know ?

vegarringdal replied:
btw, tested the material index part... we could have own material per group... but ended up with own draw call per group if I did :-(

gkjohnson replied:
This is more of a three.js question rather than a bvh question but if you have multiple groups on a mesh then they are always rendered with a different draw call for each one even if the same material is used.

gkjohnson commented:
Thanks for the contribution! I'll get a new release out this week.
